src/host/rcp_host.cpp

```diff
@@ -886,6 +886,18 @@ void RcpHost::AddEphemeralKeyStateChangedCallback(EphemeralKeyStateChangedCallba
     mEphemeralKeyStateChangedCallbacks.push_back(aCallback);
 }
 
+void RcpHost::SetBorderAgentVendorTxtData(const std::vector<uint8_t> &aVendorTxtData)
+{
+    VerifyOrExit(mInstance != nullptr);
+
+#if OTBR_ENABLE_BORDER_AGENT && !OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE
```

Comment:
I suggest we guard this for the entire `ThreadHost::SetBorderAgentVendorTxtData` interface and the implementations in RcpHost and NcpHost. In this way we know that this method is only required when `OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE` is false and we are using OT's implementation.

And this also avoids entering an empty function when `OTBR_ENABLE_BORDER_AGENT_MESHCOP_SERVICE` is true. Thoughts?